solidity/contracts/deposit/Deposit.sol

```diff
@@ -113,10 +118,10 @@ contract Deposit is DepositFactoryAuthority {
         uint256 _n,
         uint256 _lotSizeSatoshis
     ) public onlyFactory payable returns (bool) {
-        self.TBTCSystem = _TBTCSystem;
-        self.TBTCToken = _TBTCToken;
-        self.TBTCDepositToken = _TBTCDepositToken;
-        self.FeeRebateToken = _FeeRebateToken;
+        self.tbtcSystem = ITBTCSystem(_TBTCSystem);
+        self.tbtcToken = TBTCToken(_TBTCToken);
+        self.tbtcDepositToken = IERC721(_TBTCDepositToken);
+        self.feeRebateToken = FeeRebateToken(_FeeRebateToken);
```

Comment:
Can we declare the actual function parameters to have these types?

Reply:
Yes we can, It doesn't really make a difference though

Reply:
Will change anyway, might look a little cleaner